lua/cfc_notifications/shared/sh_presets.lua

```diff
@@ -23,26 +23,46 @@ CFCNotifications.registerNotificationType( "Buttons", function( CONTEXT )
     -- These require player input, so are more important than other notifications
     CONTEXT._priority = CFCNotifications.PRIORITY_NORMAL
 
-    local btnH = 30
+    local btnH = 45
     local btnBottomMargin = 10
+
     CONTEXT:SetExtraHeight( btnH + btnBottomMargin - 20 )
 
     function CONTEXT:_addDefaultButtons()
         self:AddButton( "Yes", Color( 0, 255, 0 ), true )
         self:AddButton( "No", Color( 255, 0, 0 ), false )
     end
 
-    function CONTEXT:AddButton( text, col, ... )
+    function CONTEXT:AddButtonAligned( text, col, alignment, ... )
         col = col or Color( 255, 255, 255 )
+        alignment = alignment or CFCNotifications.ALIGN_CENTER
+        self._curRowSize = ( self._curRowSize or 0 ) + 1
         self._buttons = self._buttons or {}
 
         table.insert( self._buttons, {
             text = text,
             color = col,
+            alignment = alignment,
             data = { ... }
         } )
     end
 
+    function CONTEXT:AddButton( text, col, ... )
+        self:AddButtonAligned( text, col, CFCNotifications.ALIGN_CENTER, ... )
+    end
+
+    function CONTEXT:NewButtonRow()
+        if table.IsEmpty( self._buttons ) then return end
+        if self._curRowSize < 1 then return end
+
+        self._buttons[#self._buttons].startNewRow = true
```

Comment:
Not sure I like this method of storing this sort of "new line" flag on the button before, rather than say having a multidim table of buttons like below
```lua
buttons = {
    { button1, button2 },
    { button3, button3 }
}
```
Using this flag makes it real hard to work out where a button is with iterating the structure. It also means you have to keep track of x and y in your populate loop, rather than simply using table indexes.
With this method as well, the buttons structure better represents the button positions, and makes it easier to work with this code later. Also means you don't need this rowSizes table, so cleans up the code a bit.

Reply:
Certainly, though that would necessitate restructuring the internal button structure and it would pair well with built in functions to edit the pre-existing button matrix. I feel all that would work better as a separate pr, honestly.

Reply:
Changes that improve the code shouldn't be moved to new PR's, only new functionality (imo)

Reply:
It *would* add new functionality, that's the thing.

Reply:
The backend structure change doesn't need to add new functionality, only the possibility for it. CHanging the structure here means the future PR that might be made to add said functionality doesn't need a full restructure, possibly from someone not familiar with the codebase.
Also, if its a separate PR, we're allowing code that's harder to read than it needs to be into master. Saying we'll make a PR to change that later is techdebt, and we should avoid that wherever possible.